File: 066_plus_one.java
class Solution {
    public int[] plusOne(int[] digits) {
        List<Integer> result = new ArrayList<>();
        for (int i: digits) result.add(i);

        for (int i = result.size() - 1; i >= 0; i--){
            int digit = result.get(i);
            if (digit < 9){
                result.set(i, digit + 1);
                break;
            }else{
                result.set(i, 0);
            }
        }
        if (result.get(0) == 0){
            result.add(0);
            result.set(0, 1);
        }

        int[] list = new int[result.size()];
        for (int i = 0; i < result.size(); i++) list[i] = result.get(i);
        return list;
    }
}

File: python/066_plus_one.py
class Solution:
    def plusOne2(self, digits: List[int]) -> List[int]:
        carry = 0
        digits[len(digits) - 1] += 1
        for i in range(len(digits) - 1, -1, -1):
            digits[i] += carry
            carry = 0
            if digits[i] > 9:
                carry = int(digits[i] / 10)
                digits[i] %= 10
        if carry > 0:
            digits = [carry] + digits
        return digits

    # other solution
    def plusOne(self, digits: List[int]) -> List[int]:
        sum = 0
        for i in digits:
            sum = sum * 10 + i
        return [int(x) for x in str(sum + 1)]


